lib/vimeo.js

```diff
@@ -472,6 +472,74 @@ Vimeo.prototype.upload = function (
   })
 }
 
+/**
+ * Upload a file with a File Object.
+ *
+ * This should be used to upload file object. If you want a form for your site to upload direct to
+ * Vimeo, you should look at the `POST /me/videos` endpoint.
+ *
+ * https://developer.vimeo.com/api/endpoints/videos#POST/users/{user_id}/videos
+ *
+ * @param {File}      file              File you wish to upload.
+ * @param {object=}   params            Parameters to send when creating a new video (name,
+ *                                      privacy restrictions, etc.). See the API documentation for
+ *                                      supported parameters.
+ * @param {Function}  completeCallback  Callback to be executed when the upload completes.
+ * @param {Function}  progressCallback  Callback to be executed when upload progress is updated.
+ * @param {Function}  errorCallback     Callback to be executed when the upload returns an error.
+ */
+Vimeo.prototype.uploadFile = function(
+  file,
+  params,
+  completeCallback,
+  progressCallback,
+  errorCallback
+) {
+  var _self = this
+
+  if (typeof params === 'function') {
+    errorCallback = progressCallback
+    progressCallback = completeCallback
+    completeCallback = params
+    params = {}
+  }
+
+  var fileSize = file.size
```

Comment:
To eliminate a lot of boilerplate and duplicated functionality, we could probably merge this line with line 436 and if `file` is a file object/blob, run this, else pass it through `fs.statSync`.

Reply:
Also if you don't mind, could you also mimic the same behavior within `.replace` so it can accept either a file path or a blob?

Reply:
@erunion for sure sr, I will make this change. :)

Reply:
> Also if you don't mind, could you also mimic the same behavior within `.replace` so it can accept either a file path or a blob?

@nicolastakashi I just noticed that you forgot to include this one on your change.

Also, to anyone, is there any way I could change my local copy to work while waiting for the PR to get merged?